diff --git a/dist/core.cjs b/dist/core.cjs
index eabcb539b37438c6c3581c0ac471a4f7ba334479..6fb23cd2c674eaf0c75fa5c8691797c568f1bae4 100644
--- a/dist/core.cjs
+++ b/dist/core.cjs
@@ -1022,8 +1022,8 @@ var NodeUserAuthorization = class {
     const address = this._address;
     const chainId = this._chainId;
     await this.resolveTinyCloudHostsForSignIn(address, chainId);
-    const keyId = `session-${Date.now()}`;
-    this.sessionManager.renameSessionKeyId(this.activeSessionKeyId, keyId);
+    const keyId = `session-${Date.now()}-${Math.random().toString(36).slice(2)}`;
+    this.sessionManager.createSessionKey(keyId);
     this.activeSessionKeyId = keyId;
     const jwkString = this.sessionManager.jwk(keyId);
     if (!jwkString) {
diff --git a/dist/core.js b/dist/core.js
index d5fe100381fcbaef99e7ccbe9f3abdacd35a4247..1f3ec5495ae7c9b59f873587fccfe8346cd84db5 100644
--- a/dist/core.js
+++ b/dist/core.js
@@ -948,8 +948,8 @@ var NodeUserAuthorization = class {
     const address = this._address;
     const chainId = this._chainId;
     await this.resolveTinyCloudHostsForSignIn(address, chainId);
-    const keyId = `session-${Date.now()}`;
-    this.sessionManager.renameSessionKeyId(this.activeSessionKeyId, keyId);
+    const keyId = `session-${Date.now()}-${Math.random().toString(36).slice(2)}`;
+    this.sessionManager.createSessionKey(keyId);
     this.activeSessionKeyId = keyId;
     const jwkString = this.sessionManager.jwk(keyId);
     if (!jwkString) {
diff --git a/dist/index.cjs b/dist/index.cjs
index 3d5d14aee538ffbc67e00a3d53b79129cf673f55..8e7cc5f2a362db9ad161fd4e1d1ef0917379183c 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -18062,8 +18062,8 @@ var NodeUserAuthorization = class {
     const address = this._address;
     const chainId = this._chainId;
     await this.resolveTinyCloudHostsForSignIn(address, chainId);
-    const keyId = `session-${Date.now()}`;
-    this.sessionManager.renameSessionKeyId(this.activeSessionKeyId, keyId);
+    const keyId = `session-${Date.now()}-${Math.random().toString(36).slice(2)}`;
+    this.sessionManager.createSessionKey(keyId);
     this.activeSessionKeyId = keyId;
     const jwkString = this.sessionManager.jwk(keyId);
     if (!jwkString) {
diff --git a/dist/index.js b/dist/index.js
index 073b90baf18e65586af053d1d4e0368d6f2c2807..6afbc9334265b115e6f68019eed7a05f3369bc8e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -18033,8 +18033,8 @@ var NodeUserAuthorization = class {
     const address = this._address;
     const chainId = this._chainId;
     await this.resolveTinyCloudHostsForSignIn(address, chainId);
-    const keyId = `session-${Date.now()}`;
-    this.sessionManager.renameSessionKeyId(this.activeSessionKeyId, keyId);
+    const keyId = `session-${Date.now()}-${Math.random().toString(36).slice(2)}`;
+    this.sessionManager.createSessionKey(keyId);
     this.activeSessionKeyId = keyId;
     const jwkString = this.sessionManager.jwk(keyId);
     if (!jwkString) {
